infrastructure/Taskfile.yml

```diff
@@ -498,16 +499,46 @@ tasks:
       - sh: "[ ! -z {{.VARIABLE_TYPE}} ]"
         msg: "Missing VARIABLE_TYPE"
 
-    lagoon:project:set:github-registry-credentials:
+    lagoon:ensure:environment-variable:
+      desc: |
+        Ensures (creates or updates) the state of a particular environment variable.
+        Based on this mutation resolver: https://github.com/uselagoon/lagoon/blob/1ef8e57ffbc969a308e9a16bb53929edac562a4d/services/api/src/resources/env-variables/resolvers.ts#L300
       vars:
-        # VARIABLE_TYPE_ID is the PROJECT_ID. If we are given a PROJECT_NAME, lookup the ID
-        VARIABLE_TYPE_ID:
-          sh:
-            if [ ! -z {{.PROJECT_ID}} ]; then
-              echo "{{.PROJECT_ID}}";
+        ENVIRONMENT_NAME_INPUT:
+          sh: |
+            if [ -z "{{.ENVIRONMENT_NAME}}" ]; then
+              echo -n "";
             else
-              lagoon get project --project "{{.PROJECT_NAME}}" --output-json | jq '.data[0].id' --raw-output;
+              echo -n 'environment: \"{{.ENVIRONMENT_NAME}}\",';
             fi
+      cmds:
+        - task: lagoon:run-mutation
+          vars:
+            MUTATION: |
+              mutation ensureEnvironmentVariable {
+                addOrUpdateEnvVariableByName(
+                  input:{
+                    project:\"{{.PROJECT_NAME}}\",{{.ENVIRONMENT_NAME_INPUT}}
+                    scope:{{.VARIABLE_SCOPE}},
+                    name:\"{{.VARIABLE_NAME}}\",
+                    value:\"{{.VARIABLE_VALUE}}\"
+                  }
+                ) {
+                  id
+                }
+              }
+      preconditions:
+      - sh: "[ ! -z {{.VARIABLE_VALUE}} ]"
+        msg: "Missing VARIABLE_VALUE"
+      - sh: "[ ! -z {{.VARIABLE_NAME}} ]"
+        msg: "Missing VARIABLE_NAME"
+      - sh: "[ ! -z {{.VARIABLE_SCOPE}} ]"
+        msg: "Missing VARIABLE_SCOPE"
+      - sh: "[ ! -z {{.PROJECT_NAME}} ]"
+        msg: "Missing PROJECT_NAME"
```

Comment:
In the latest versions of Task you can use the `requires` property instead. I think it is a bit more clean:

https://github.com/danskernesdigitalebibliotek/dpl-cms/blob/9716206769278d6dc6781fe554b9a94f7a93e9f8/task/Taskfile.translations.yml#L37-L38